src/modules/pmc/src/lib/PmcBase.cpp

```diff
@@ -561,3 +599,114 @@ std::string PmcBase::Trim(const std::string& str, const std::string& trim)
 {
     return TrimStart(TrimEnd(str, trim), trim);
 }
+
+std::vector<std::string> PmcBase::ListFiles(const char* directory, const char* fileNameExtension)
+{
+    struct dirent* entry = nullptr;
+    DIR* directoryStream = nullptr;
+    int extensionLength = nullptr != fileNameExtension ? strlen(fileNameExtension) : 0;
```

Comment:
>nullptr != fileNameExtension ? strlen(fileNameExtension) : 0; [](http://example.com/codeflow?start=26&length=61)

Add parenthesis. Also consider that attempting strlen on a null string can result in crash

Reply:
I wasn't able to cause any crash. I tried passing `NULL`, and also `""` and it works correctly. When it's `NULL` then I do not use `strlen`, but I just put `0`.